packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -63,13 +63,27 @@ export class SummarizerNode implements IRootSummarizerNode {
 	 * Returns 0 if there is not yet an acked summary.
 	 */
 	public get referenceSequenceNumber() {
-		return this._latestSummary?.referenceSequenceNumber ?? 0;
+		return this._lastSummaryReferenceSequenceNumber ?? 0;
+	}
+
+	/**
+	 * returns the handle of the last successful summary of this summarizerNode in string format
+	 * (this getter is primarily only used in the test code)
+	 */
+	public get summaryHandleId(): string {
+		return this._summaryHandleId.toString();
 	}
 
 	protected readonly children = new Map<string, SummarizerNode>();
-	protected readonly pendingSummaries = new Map<string, SummaryNode>();
+	/** Key value pair of summaries submitted by this client which are not yet acked. Key is the proposalHandle and value is the summary op's referece sequence number */
+	protected readonly pendingSummaries = new Map<string, PendingSummaryInfo>();
 	protected wipReferenceSequenceNumber: number | undefined;
-	private wipLocalPaths: { localPath: EscapedPath; additionalPath?: EscapedPath } | undefined;
+	/**
+	 * True if the current node was summarized during the current summary process
+	 * This flag is used to identify scenarios where a node may not have been summarized during the summary
+	 * because of application code creating data structures for data stores which were already summarized.
+	 */
+	private wipSummary: boolean = false;
```

Comment:
Consider renaming this to something that conveys whether this node was summarizer, like `wipSummarizeCalled` or `wipWasSummarized`.